src/components/Charts/mutationsTree.js

```diff
@@ -8,33 +8,22 @@ class MutationsTree extends React.Component {
         this.state = {};
     }
 
-    createVariants(variants, level){
-        return variants.flatMap((variant) => [
+    createVariants = (variants, level) => {
+        return (variants.flatMap((variant) => [
             `${'---'.repeat(level)}variant: ${variant.name} (${variant.mutations.map(mutation=>mutation.from+mutation.position+mutation.to).join(", ")})`,
             ...this.createVariants(variant.subs, level+1)
-         ])
-    }
-
-
-    componentDidMount() {
-        let variantsTree = this.createVariants(this.props.data, 0).map(i => {
+         ])).map(i => {
```

Comment:
Is this correct? createVariants is recursive, and this therefore `<p>` escapes recursively. Is that what we want?

Reply:
yes, you are right, this might be a problem. maybe if we seperated the second map to the second function, it might by okay.